frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx

```diff
@@ -17,91 +16,90 @@ import {
+  cancel,
+  close,
+  k8sUpdate,
+  obj,
+}) => {
+  const { t } = useTranslation();
+  const [handlePromise, inProgress, errorMessage] = usePromiseHandler();
+  const [selectedApprovalStrategy, setSelectedApprovalStrategy] = React.useState(
+    getApprovalStrategy(obj),
+  );
+  const submit = React.useCallback(
+    (event: React.FormEvent<HTMLFormElement>): void => {
+      event.preventDefault();
+      const updatedObj = _.cloneDeep(obj);
+      if (referenceFor(updatedObj) === referenceForModel(SubscriptionModel)) {
+        (updatedObj as SubscriptionKind).spec.installPlanApproval = selectedApprovalStrategy;
+      } else if (referenceFor(updatedObj) === referenceForModel(InstallPlanModel)) {
+        (updatedObj as InstallPlanKind).spec.approval = selectedApprovalStrategy;
+      }
+      handlePromise(k8sUpdate(modelFor(referenceFor(obj)), updatedObj))
+        .then(() => close?.())
+        .catch(_.noop);
+    },
+    [close, handlePromise, k8sUpdate, obj, selectedApprovalStrategy],
+  );
 
-  constructor(public props: InstallPlanApprovalModalProps) {
-    super(props);
-
-    this.state.selectedApprovalStrategy = getApprovalStrategy(props);
-  }
-
-  private submit(event): void {
-    event.preventDefault();
-
-    const updatedObj = _.cloneDeep(this.props.obj);
-    if (referenceFor(updatedObj) === referenceForModel(SubscriptionModel)) {
-      (updatedObj as SubscriptionKind).spec.installPlanApproval = this.state.selectedApprovalStrategy;
-    } else if (referenceFor(updatedObj) === referenceForModel(InstallPlanModel)) {
-      (updatedObj as InstallPlanKind).spec.approval = this.state.selectedApprovalStrategy;
-    }
-    this.handlePromise(this.props.k8sUpdate(modelFor(referenceFor(this.props.obj)), updatedObj))
-      .then(() => this.props.close())
-      .catch((err) => this.setState({ errorMessage: err }));
-  }
-
-  render() {
-    return (
-      <form onSubmit={this.submit.bind(this)} name="form" className="modal-content">
-        <ModalTitle className="modal-header">Change Update Approval Strategy</ModalTitle>
-        <ModalBody>
-          <div className="co-m-form-row">
-            <p>What strategy is used for approving updates?</p>
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle className="modal-header">{t('olm~Change Update Approval Strategy')}</ModalTitle>
+      <ModalBody>
+        <div className="co-m-form-row">
+          <p>{t('olm~What strategy is used for approving updates?')}</p>
+        </div>
+        <div className="co-m-form-row row">
+          <div className="col-sm-12">
+            <RadioInput
+              onChange={(e) => setSelectedApprovalStrategy(e.target.value)}
+              value={InstallPlanApproval.Automatic}
+              checked={selectedApprovalStrategy === InstallPlanApproval.Automatic}
+              title={t(`olm~${InstallPlanApproval.Automatic}`)}
```

Comment:
I'd do it this way when possible.

```suggestion
              title={t('olm~Automatic')}
```